staging/src/k8s.io/code-generator/cmd/applyconfiguration-gen/generators/util.go

```diff
@@ -133,16 +141,67 @@ func (g *utilGenerator) GenerateType(c *generator.Context, _ *types.Type, w io.W
 	sort.Sort(groupSort(groups))
 
 	m := map[string]interface{}{
+		"applyConfiguration":     applyConfiguration,
+		"fmtErrorf":              fmtErrorf,
 		"groups":                 groups,
+		"internalParser":         types.Ref(path.Join(g.outputPackage, "internal"), "Parser"),
+		"runtimeObject":          runtimeObject,
 		"schemeGVs":              schemeGVs,
 		"schemaGroupVersionKind": groupVersionKind,
-		"applyConfiguration":     applyConfiguration,
+		"toOpenAPIName":          toOpenAPIName,
+		"typedValue":             typedValue,
+		"unstructured":           unstructured,
+		"validationOptions":      validationOptions,
+		"gvkToOpenAPIType":       g.typeModels.gvkToOpenAPIType,
 	}
 	sw.Do(forKindFunc, m)
+	sw.Do(fakeTypeConverter, m)
 
 	return sw.Error()
 }
 
+var fakeTypeConverter = `
+type FakeTypeConverter struct {}
+
+func (tc FakeTypeConverter) ObjectToTyped(obj {{.runtimeObject|raw}}, opts ...{{.validationOptions|raw}}) (*{{.typedValue|raw}}, error) {
+	gvk := obj.GetObjectKind().GroupVersionKind()
+	t := {{.internalParser|raw}}().Type(openAPIName(gvk))
+	switch o := obj.(type) {
+	case *{{.unstructured|raw}}:
+		return t.FromUnstructured(o.UnstructuredContent(), opts...)
+	default:
+		return t.FromStructured(obj, opts...)
+	}
+}
+
+func (tc FakeTypeConverter) TypedToObject(value *{{.typedValue|raw}}) ({{.runtimeObject|raw}}, error) {
+	vu := value.AsValue().Unstructured()
+	switch o := vu.(type) {
+	case map[string]interface{}:
+		return &{{.unstructured|raw}}{Object: o}, nil
+	default:
+		return nil, {{.fmtErrorf|raw}}("failed to convert value to unstructured for type %T", vu)
+	}
+}
+
+func openAPIName(kind {{.schemaGroupVersionKind|raw}}) string {
+	switch kind {
+		{{range $group := .groups -}}{{$GroupGoName := .GroupGoName -}}
+			{{range $version := .Versions -}}
+	// Group={{$group.Name}}, Version={{.Name}}
+				{{range .Resources -}}
+					{{if .HasOpenAPIName -}}
```

Comment:
IIRC ``{{if .OpenAPIName}}` would work the same way and possibly allow you to avoid this extra field

Reply:
Sweet. Trying  it out.